### PR TITLE
Verbose setting

### PR DESCRIPTION
**Pull request by gwenchee**
In this PR, I used python's logging package to add logging to ROLLO. The user enables this by adding a `-v` flag:
`python -m rollo -i <input file> -c <checkpoint file> -v `

I also addressed #33 and #9 by adding warnings loggings. The WARNING logs will appear without `-v` flag. The INFO logs will only appear with `-v` flag. 

An example ROLLO output looks like this: 
```
WARNING:root: ROLLO added default for variable: mutation_probability, default value = 0.23
WARNING:root: ROLLO added default for variable: mating_probability, default value = 0.47
WARNING:root: ROLLO added default for variable: mutation_operator, default value = {'operator': 'mutPolynomialBounded', 'eta': 0.23, 'indpb': 0.23}
WARNING:root: ROLLO added default for variable: mating_operator, default value = {'operator': 'cxBlend', 'alpha': 0.46}
Entering generation 0...
WARNING:root: parallel method = job_control
INFO:root: Solver: openmc, Input Script Runtime: 21.1 seconds
INFO:root: Solver: openmc, Execute 0 Runtime: 13.73 seconds
INFO:root: Solver: openmc, Output Script Runtime: 21.23 seconds
INFO:root: Generation: 0, Evaluation Total Runtime: 56.13 seconds
WARNING:root: 3 out of 4 inds were constrained
                                                              oup                                                           ind
                        ------------------------------------------------------------------------------- --------------------------------------------
time    gen     evals   avg                     std     min                     max                     avg             min             max
56.1503 0       4       [6.75119225 1.29395975] [0. 0.] [6.75119225 1.29395975] [6.75119225 1.29395975] [6.75119225]    [6.75119225]    [6.75119225]
Entering generation 1...
INFO:root: Solver: openmc, Input Script Runtime: 20.27 seconds
INFO:root: Solver: openmc, Execute 0 Runtime: 12.46 seconds
INFO:root: Solver: openmc, Output Script Runtime: 20.23 seconds
INFO:root: Generation: 1, Evaluation Total Runtime: 53.01 seconds
WARNING:root: 0 out of 4 inds were constrained
                                                              oup                                                           ind
                        ------------------------------------------------------------------------------- --------------------------------------------
time    gen     evals   avg                     std     min                     max                     avg             min             max
56.1503 0       4       [6.75119225 1.29395975] [0. 0.] [6.75119225 1.29395975] [6.75119225 1.29395975] [6.75119225]    [6.75119225]    [6.75119225]
109.16  1       3       [6.75119225 1.29395975] [0. 0.] [6.75119225 1.29395975] [6.75119225 1.29395975] [6.75119225]    [6.75119225]    [6.75119225]
rollo Simulation Completed!
Total time in simulation 109.16 seconds
```

### CODE REVIEW

**Comment by gwenchee** [comment]
Any takers @nsryan2 @abachma2 @samgdotson @smpark7 @LukeSeifert @yardasol

File: README.md [hunk 1]
```diff
@@ -11,19 +11,22 @@ Documentation on the usage of ROLLO is hosted at: https://gwenchee.github.io/rol
...
 ## Running ROLLO 
-`python -m rollo -i <input file> -c <checkpoint file>`
+`python -m rollo -i <input file> -c <checkpoint file> -v `
 
 Command line flags: 
 | Flag | Description | Mandatory ? |
 | ----------- | ----------- | ----------- |
 | -i | name of input file | Yes |
 | -c| name of checkpoint file | No |
+| -v| verbrose output (only include the flag) | No |
```
Comment (smpark7) [inline README.md:21]:
```suggestion
| -v| verbose output (only include the flag) | No |
```

File: README.md [hunk 2]
```diff
@@ -11,19 +11,22 @@ Documentation on the usage of ROLLO is hosted at: https://gwenchee.github.io/rol
...
 
 The checkpoint 
file holds the results from the ROLLO simulation and also acts 
 as a restart 
file. Thus, if a ROLLO simulation ends prematurely, the checkpoint 
 file can be used to restart the code from the most recent population and 
 continue the simulation.
 
+The verbrose flag (-v) enables ROLLO to output extra information for the user.
```
Comment (smpark7) [inline README.md:28]:
```suggestion
The verbose flag (-v) enables ROLLO to output extra information for the user. 
```

File: rollo/__main__.py [hunk 3]
```diff
@@ -5,22 +5,27 @@
 
 def main():
     argv = sys.argv[1:]
-    msg = "python rollo -i <inputfile> -c <checkpoint file> "
+    msg = "python rollo -i <inputfile> -c <checkpoint file> -v "
```
Comment (smpark7) [inline rollo/__main__.py:8]: Does rollo support `--help`? If not, is there documentation for rollo command options other than the github readme?

File: rollo/__main__.py [hunk 4]
```diff
@@ -5,22 +5,27 @@
...
-                new_run = executor.Executor(input_file=opts_dict["-i"])
-            new_run.execute()
-        if len(opts) == 0:
-            raise Exception("To run rollo: " + msg)
+        if "-c" in opts_dict:
+            cp_file = opts_dict['-c']
+        else:
+            cp_file = None
+        if "-v" in opts_dict:
+            verbrose = True
```
Comment (smpark7) [inline rollo/__main__.py:21]:
```suggestion
            verbose = True
```

File: rollo/__main__.py [hunk 5]
```diff
@@ -5,22 +5,27 @@
...
+        else:
+            verbrose = False
```
Comment (smpark7) [inline rollo/__main__.py:23]:
```suggestion
            verbose = False
```

File: rollo/__main__.py [hunk 6]
```diff
@@ -5,22 +5,27 @@
...
+        new_run = executor.Executor(
+            input_file=opts_dict['-i'],
+            checkpoint_file=cp_file,
+            verbrose=verbrose)
```
Comment (smpark7) [inline rollo/__main__.py:27]:
```suggestion
            verbose=verbose)
```

File: rollo/executor.py [hunk 7]
```diff
@@ -36,12 +37,15 @@ class Executor(object):
         Name of input file
     checkpoint_file : str
         Name of checkpoint file
+    verbrose : bool
```
Comment (smpark7) [inline rollo/executor.py:40]:
```suggestion
    verbose : bool
```

File: rollo/executor.py [hunk 8]
```diff
@@ -36,12 +37,15 @@ class Executor(object):
...
 
     """
 
-    def __init__(self, input_file, checkpoint_file=None):
+    def __init__(self, input_file, checkpoint_file=None, verbrose=False):
```
Comment (smpark7) [inline rollo/executor.py:44]:
```suggestion
    def __init__(self, input_file, checkpoint_file=None, verbose=False):
```

File: rollo/executor.py [hunk 9]
```diff
@@ -36,12 +37,15 @@ class Executor(object):
...
         self.input_file = input_file
         self.checkpoint_file = checkpoint_file
+        if verbrose:
```
Comment (smpark7) [inline rollo/executor.py:47]:
```suggestion
        if verbose:
```

File: rollo/__main__.py [hunk 10]
```diff
@@ -5,22 +5,27 @@
 
 def main():
     argv = sys.argv[1:]
-    msg = "python rollo -i <inputfile> -c <checkpoint file> "
+    msg = "python rollo -i <inputfile> -c <checkpoint file> -v "
     try:
-        opts, args = getopt.getopt(argv, "i:c:")
+        opts, args = getopt.getopt(argv, "i:c:v")
+        if len(opts) == 0:
+            raise Exception("To run rollo: " + msg)
```
Comment (smpark7) [inline rollo/__main__.py:12]:
```suggestion
            raise Exception("To run ROLLO: " + msg)
```

File: rollo/evaluation.py [hunk 11]
```diff
@@ -288,14 +300,28 @@ def run_input_and_execute_and_output_scripts(
...
                 subprocess.call(execute_input, shell=True)
+                end_time = time.time()
+                logging.info(" Solver: " +
+                             solver +
+                             ", Execute " +
+                             str(i) +
+                             " Runtime: " +
+                             str(round(end_time -
+                                       start_time, 2)) +
+                             " seconds")
```
Comment (smpark7) [inline rollo/evaluation.py:313]: It's not clear what `str(i)` refers to.

File: rollo/__main__.py (hunk 10 above)
Comment (smpark7) [reply rollo/__main__.py:12]: Also, do you need this line if you have line 30? Do they cover different fail cases?

**Review by smpark7 — CHANGES_REQUESTED** [review]
Nice work! I think your changes here will improve the user-friendliness of ROLLO. I can approve this PR after you address my review comments.

File: rollo/__main__.py (hunk 10 above)
Comment (gwenchee) [reply rollo/__main__.py]: Yeah. The first one covers if the user does not specify any flags. The second one is if they specify any flags incorrectly.

File: rollo/evaluation.py (hunk 11 above)
Comment (gwenchee) [reply rollo/evaluation.py]: I added a more explicit name: `execute_index`

Comment (gwenchee) [reply rollo/evaluation.py]: and also added the identifier that it's an execute task

**Comment by yardasol** [comment]
Sorry @gwenchee, I bricked my computer the other day so I've been getting things up and running again. In the future I should be much quicker to respond to your PRs.

**Comment by gwenchee** [comment]
No worries! I have more PRs coming up @yardasol

**Pull request merged**